tests/fortran/test_fortran_client.py

```diff
@@ -41,23 +40,30 @@ def get_test_names():
     """Obtain test names by globbing for client_test
     Add tests manually if necessary
     """
-    glob_path = osp.join(TEST_PATH, "build/client_test*")
+    glob_path = osp.join(TEST_PATH, "client_test*")
     test_names = glob(glob_path)
-    test_names = [(pytest.param(test, id=osp.basename(test)))
-                  for test in test_names if not "gpu" in test]
+    test_names = list(filter(lambda test: test.find('gpu') == -1, test_names))
+    test_names = [(pytest.param(test,
+                                id=osp.basename(test))) for test in test_names]
     return test_names
 
 
 @pytest.mark.parametrize("test", get_test_names())
-def test_fortran_client(test):
+def test_fortran_client(test, build):
     """This function actually runs the tests using the parameterization
     function provided in Pytest
 
     :param test: a path to a test to run
     :type test: str
     """
-    cmd = []
-    cmd.append(test)
+    # Build the path to the test executable from the source file name
+    # . keep only the last three parts of the path: (tests, language, basename)
+    test = "/".join(test.split("/")[-3:])
```

Comment:
Same as above about rewriting using pathlib

Reply:
No clean way to do thus via pathlib since we have to keep three levels of path